Foundation/FileHandle.swift

```diff
@@ -405,6 +280,213 @@ open class FileHandle : NSObject, NSSecureCoding {
     public static var supportsSecureCoding: Bool {
         return true
     }
+    
+    private var _isPlatformHandleValid: Bool {
```

Comment:
Might be a good idea to move this up into the Windows/Non-Windows  FD & handle blocks at the top to keep all the fd and fd checks in one place.